### DISCUSSION TITLE
Datatable Slow Rendering

### DISCUSSION THREAD

**Discussion by veladzic** [primefaces/community#114] [PrimeReact] [answered]
Datatable renders really slow and typing in the search term is being really laggy, when the row count is more than 200.
I don't know if it's some random bug in my code or just the way i am filtering the table
Displaying 20 rows and searching, doesn't create any issues.

This is how my code looks like:
using a global filter like this:
```
  const onGlobalFilterChange = (e) => {
    let value = e;

    let _filters1 = { ...filters };
    _filters1["global"].value = value;

    setFilters(_filters1);
    setGlobalFilterValue(value);
  };
```

which gets triggered by:
```
 <InputText
            value={globalFilterValue}
            onChange={(e) => onGlobalFilterChange(e.target.value)}
            placeholder="Suche"
            style={{ width: "20em" }}
          />
```

and initiated like this:

```
  const initFilters = () => {
    var generatedFilterSettings = [];
    columns.forEach((element) => {
      generatedFilterSettings[element.field] = {
        operator: FilterOperator.AND,
        constraints: [{ value: null, matchMode: FilterMatchMode.STARTS_WITH }],
      };
    });
    generatedFilterSettings["global"] = {
      value: null,
      matchMode: FilterMatchMode.CONTAINS,
    };
    setFilters(generatedFilterSettings);
  };

```

datatable:
```
      <DataTable
        expandedRows={expandedRows}
        ref={dt}
        paginator
        rows={rowCount}
        reorderableColumns
        stripedRows
        scrollHeight="flex"
        dragSelection
        selectionMode={selectionMode}
        selection={selectedMultiple}
        sortMode="multiple"
        dataKey="TIDNR"
        resizableColumns
        columnResizeMode={customTableWidth}
        onContextMenu={(e) => cm.current.show(e.originalEvent)}
        value={data}
        filters={filters}
        scrollable
        filterDisplay="menu"
        loading={loading}
        responsiveLayout="scroll"
        globalFilterFields={["TIDNR"]}
        header={header}
        emptyMessage="Keine PCs."
        scrollDirection="vertical"
        onRowToggle={(e) => setExpandedRows(e.data)}
        rowExpansionTemplate={(e) => {
          return Computerdetails(e, translationTable);
        }}
        onRowSelect={(ev) => {
          startPing(
            ev,
            data,
            (e) => {
              setData(e);
            },
            selectedMultiple
          );
        }}
        onRowDoubleClick={(ev) => {
          startNetsupport([ev.data]);
        }}

        onSelectionChange={(e) => {
          if (e.value instanceof Array) {
            setScriptlaucherListGlobal(e.value);
            setSelectedMultiple(e.value);
          } else if(e.value !== null) {
            setScriptlaucherListGlobal([e.value]);
            setSelectedMultiple([e.value]);
          } else{
            setScriptlaucherListGlobal([]);
            setSelectedMultiple([]);
          }
        }}
        onContextMenuSelectionChange={(e) => {
          if (!selectedMultiple.includes(e.value)) {
            setSelectedMultiple([...selectedMultiple, e.value]);
          }
        }}
       
      >
        <Column
        columnKey="1"
          selectionMode={selectionMode}
          style={{ maxWidth: "40px", height: "10px" }}
        ></Column>
        <Column columnKey="2" expander style={{ maxWidth: "60px", height: "10px" }} />
        {columnComponents}
      </DataTable>
```

What to do make the search going faster, thanks for help :)

**Comment by melloware** [comment]
Performance enhancements have been made for 10.0.0 which is hopefully coming out tomorrow: https://github.com/primefaces/primereact/issues/4768

So after 10.0.0 comes out can you report back if its better?  Also make sure you are not in Strict Mode as duplicate re-renders in StrictMode make everything slower.

However if you are displaying 200 rows on screen they are all being rendered whereas 20 rows is a lot less to re-render.

**Reply by veladzic** [reply]
okay thanks, i will report back! Strict Mode is disabled.  https://github.com/primefaces/primereact/issues/4768 is describing my issue very well.

**Reply by melloware** [reply]
OK good when 10.0.0 comes out can you retest and report back. it is supposed to be out today or tomorrow.

**Comment by cagataycivici** [comment]
For large datasets, we suggest lazy loading but 200 should be fine for local data, could you please provide a reproducer via our template;

https://codesandbox.io/s/primereact-test-forked-bbns8k

That would really help to pinpoint the case.

**Reply by veladzic** [reply]
i tried do reproduce the issue here:
https://codesandbox.io/s/primereact-test-forked-hw8rwr?file=/src/index.js

my own dataset is a bit different but it's the same principle.
the difference is that the data that is being searched starts with 5 similar letters, and the last 5 letters are different.
now when you start to type in the first 5 letters the table is lagging as there are many matches. this is something you can observe in the codesandbox i provided. 

lazy loading would be fine but it takes longer too. my use case is a bit different, as i need the data when filtering instantly. i am loading ~ 3000 rows - when using pagination and an max rowcount of 20, filtering the data is really fast and happens instantly. but when increasing the row count to be displayed to more than 200, with pagination activated, the input starts lagging.

**Reply by melloware** [reply]
Thanks for the reproducer. I am looking forward to trying this out with 10.0.0.  If it is still an issue I will have you open a ticket with this reproducer against 10.0.0

**Reply by melloware** [reply]
10.0.0 is out if you want to re-test.

**Reply by veladzic** [reply]
i tested it, but unfortunately it is behaving in the same way as before.

**Reply by melloware** [reply, marked as answer]
Ok please comment on this ticket and add your reproducer and scenario. https://github.com/primefaces/primereact/issues/2979

It's related to re renders
